application/src/main/java/org/thingsboard/server/service/security/auth/TokenOutdatingService.java

```diff
@@ -17,49 +17,43 @@
 
 import io.jsonwebtoken.Claims;
 import lombok.RequiredArgsConstructor;
-import org.springframework.cache.Cache;
-import org.springframework.cache.CacheManager;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
-import org.thingsboard.server.common.data.CacheConstants;
+import org.springframework.transaction.event.TransactionalEventListener;
+import org.thingsboard.server.cache.usersUpdateTime.UsersUpdateTimeCacheEvictEvent;
 import org.thingsboard.server.common.data.id.UserId;
 import org.thingsboard.server.common.data.security.event.UserAuthDataChangedEvent;
 import org.thingsboard.server.common.data.security.model.JwtToken;
 import org.thingsboard.server.config.JwtSettings;
+import org.thingsboard.server.dao.entity.AbstractCachedEntityService;
 import org.thingsboard.server.service.security.model.token.JwtTokenFactory;
 
-import javax.annotation.PostConstruct;
+import java.util.HashMap;
 import java.util.Optional;
 
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 @Service
 @RequiredArgsConstructor
-public class TokenOutdatingService {
-    private final CacheManager cacheManager;
+public class TokenOutdatingService extends AbstractCachedEntityService<UserId, HashMap<String, Long>, UsersUpdateTimeCacheEvictEvent> {
     private final JwtTokenFactory tokenFactory;
     private final JwtSettings jwtSettings;
-    private Cache usersUpdateTimeCache;
-
-    @PostConstruct
-    protected void initCache() {
-        usersUpdateTimeCache = cacheManager.getCache(CacheConstants.USERS_UPDATE_TIME_CACHE);
-    }
 
     @EventListener(classes = UserAuthDataChangedEvent.class)
     public void onUserAuthDataChanged(UserAuthDataChangedEvent event) {
-        usersUpdateTimeCache.put(toKey(event.getUserId()), event.getTs());
+        processUserSessions(event);
     }
 
     public boolean isOutdated(JwtToken token, UserId userId) {
         Claims claims = tokenFactory.parseTokenClaims(token).getBody();
         long issueTime = claims.getIssuedAt().getTime();
 
-        return Optional.ofNullable(usersUpdateTimeCache.get(toKey(userId), Long.class))
```

Comment:
Check that userId is not outdated, and then check is this concrete sessionId is not outdated